implementation/fault-tolerance/src/main/java/io/smallrye/faulttolerance/CdiLogger.java

```diff
@@ -23,9 +23,13 @@
     @Message(id = 2, value = "Multiple circuit breakers have the same name '%s': %s")
     DefinitionException multipleCircuitBreakersWithTheSameName(String name, Set<String> useSites);
 
-    @Message(id = 3, value = "Backoff annotation @%s present on method '%s', but @Retry is missing")
-    DefinitionException backoffOnMethodWithoutRetry(String backoffAnnotation, MethodDescriptor method);
+    @Message(id = 3, value = "Backoff annotation @%s present on '%s', but @Retry is missing")
+    DefinitionException backoffAnnotationWithoutRetry(String backoffAnnotation, MethodDescriptor method);
 
-    @Message(id = 4, value = "Backoff annotation @%s present on class '%s', but @Retry is missing")
-    DefinitionException backoffOnClassWithoutRetry(String backoffAnnotation, Class<?> clazz);
+    DefinitionException backoffAnnotationWithoutRetry(String backoffAnnotation, Class<?> clazz);
+
+    @Message(id = 4, value = "Both @Blocking and @NonBlocking present on '%s'")
+    DefinitionException bothBlockingNonBlockingPresent(MethodDescriptor method);
+
+    DefinitionException bothBlockingNonBlockingPresent(Class<?> clazz);
```

Comment:
This would be a breaking change, but messages with code 3 and above are not present in a release yet, so it's OK.